### PR TITLE
API-6683 - Upgrade to Scala 2.13

### PR DESCRIPTION
**Pull request by peteslater-ee**

### CODE REVIEW

File: acceptance/uk/gov/hmrc/apisubscriptionfields/ApiSubscriptionFieldsHappySpec.scala [hunk 1]
```diff
@@ -210,7 +210,7 @@ class ApiSubscriptionFieldsHappySpec extends AcceptanceTestSpec
       val result: Option[Future[Result]] = route(app, request)
 
       Then(s"a response with a 200 status is received")
-      result shouldBe 'defined
+      result shouldBe Symbol("defined")
```
[the remark this answers is not in the document]
Comment (AndySpaven) [reply acceptance/uk/gov/hmrc/apisubscriptionfields/ApiSubscriptionFieldsHappySpec.scala:213]: the tick version must be super old scalatest

**Review by AndySpaven — CHANGES_REQUESTED** [review]

File: build.sbt [hunk 2]
```diff
@@ -74,7 +68,13 @@ lazy val microservice = Project(appName, file("."))
...
-  .settings(scalacOptions ++= Seq("-Ypartial-unification"))
+  .settings(
+    scalacOptions ++= Seq(
+    "-Wconf:cat=unused&src=.*RoutesPrefix\\.scala:s",
+    "-Wconf:cat=unused&src=.*Routes\\.scala:s",
+    "-Wconf:cat=unused&src=.*ReverseRoutes\\.scala:s"
+    )
+  )
 
 lazy val acceptanceTestSettings =
```
Comment (AndySpaven) [inline build.sbt:79]: Can we add a section

`  commands += Command.command("testAll") { state =>
      "test" :: "it:test" :: state
  }
  `

Comment (AndySpaven) [reply build.sbt:79]: I think this needs test :: it:test :: acceptance:test if this has all 3

Comment (peteslater-ee) [reply build.sbt]: It just needed 2  test and acceptanceTest

**Pull request merged**